app/src/main/java/org/simple/clinic/sync/indicator/SyncIndicatorUpdate.kt

```diff
@@ -4,15 +4,78 @@ import com.spotify.mobius.Next
 import com.spotify.mobius.Next.noChange
 import com.spotify.mobius.Update
 import org.simple.clinic.mobius.next
+import org.simple.clinic.sync.LastSyncedState
+import org.simple.clinic.sync.SyncInterval
+import org.simple.clinic.sync.SyncProgress.FAILURE
+import org.simple.clinic.sync.SyncProgress.SUCCESS
+import org.simple.clinic.sync.SyncProgress.SYNCING
+import org.simple.clinic.sync.indicator.SyncIndicatorState.ConnectToSync
+import org.simple.clinic.sync.indicator.SyncIndicatorState.SyncPending
+import org.simple.clinic.sync.indicator.SyncIndicatorState.Synced
+import org.simple.clinic.sync.indicator.SyncIndicatorState.Syncing
+import org.threeten.bp.Duration
+import org.threeten.bp.Instant
+import java.util.concurrent.TimeUnit
 
 class SyncIndicatorUpdate : Update<SyncIndicatorModel, SyncIndicatorEvent, SyncIndicatorEffect> {
   override fun update(model: SyncIndicatorModel, event: SyncIndicatorEvent):
       Next<SyncIndicatorModel, SyncIndicatorEffect> {
 
     return when (event) {
-      is LastSyncedStateFetched -> next(model.lastSyncedStateChanged(event.lastSyncState))
-      is DataForSyncIndicatorStateFetched -> noChange()
-      is IncrementTimerTick -> noChange()
+      is LastSyncedStateFetched -> next(model.lastSyncedStateChanged(event.lastSyncState), FetchDataForSyncIndicatorState)
+      is DataForSyncIndicatorStateFetched -> updateSyncIndicatorState(model, event.currentTime, event.syncIndicatorFailureThreshold)
+      is IncrementTimerTick -> next(model.syncIndicatorSyncedTimerIncremented(event.minutes))
     }
   }
+
+  private fun updateSyncIndicatorState(
+      model: SyncIndicatorModel,
+      currentTime: Instant,
+      failureThreshold: Duration
+  ): Next<SyncIndicatorModel, SyncIndicatorEffect> {
+    if (model.lastSyncedState == null || model.lastSyncedState.isEmpty())
+      return noChange()
+
+    val syncIndicatorState = when (model.lastSyncedState.lastSyncProgress!!) {
+      SUCCESS, FAILURE -> syncIndicatorState(model.lastSyncedState, failureThreshold, currentTime)
+      SYNCING -> Syncing
+    }
+
+    return when (syncIndicatorState) {
+      is Synced -> next(model.syncIndicatorStateChanged(syncIndicatorState), StartSyncedStateTimer(
+          intervalAmount = 1,
+          timeUnit = TimeUnit.MINUTES
+      ))
+      else -> next(model.syncIndicatorStateChanged(syncIndicatorState))
+    }
+  }
+
+  //TODO: Refactor this method after migration to Mobius is complete.
+  // This method is copied as is from `SyncIndicatorController` for migration purpose.
+  private fun syncIndicatorState(
+      syncState: LastSyncedState,
+      maxIntervalSinceLastSync: Duration,
+      currentTime: Instant
+  ): SyncIndicatorState {
+    val timestamp = syncState.lastSyncSucceededAt ?: return SyncPending
```

Comment:
I prefer using the elvis operator to provide an alternate value, but it seems like we're using it here to provide an alternate code path which is a little hard to understand at parse. What do you think of making this early return explicit by moving it to an `if` block?

Reply:
I have added a comment here saying that this will be refactored later. I wanted to keep this method easy to review by not changing it at all from `Controller` class.
I will make this change in cleanup PR.